docs/source/llms/rag/notebooks/question-generation-retrieval-evaluation.ipynb

```diff
@@ -504,95 +573,130 @@
+    "    \n",
+    "Paragraph: {content}\n",
     "\n",
-    "    response = openai.ChatCompletion.create(**params)\n",
-    "    response_queries = response.choices[0].message.content\n",
+    "Please call the submit_function function to submit the generated question and answer.\n",
+    "    \"\"\"\n",
     "\n",
-    "    question_list = []\n",
-    "    for q in response_queries.splitlines():\n",
-    "        q = \" \".join(q.split()[1:])\n",
-    "        question_list.append(q)\n",
+    "    messages = [{\"role\": \"user\", \"content\": prompt}]\n",
     "\n",
-    "    queries.append({\"chunk\": chunk, \"questions\": question_list})\n",
+    "    submit_function = {\n",
+    "        \"name\": \"submit_function\",\n",
+    "        \"description\": \"Call this function to submit the generated question and answer.\",\n",
+    "        \"parameters\": {\n",
+    "            \"type\": \"object\",\n",
+    "            \"properties\": {\n",
+    "                \"question\": {\n",
+    "                    \"type\": \"string\",\n",
+    "                    \"description\": \"The question asking for the key information in the given paragraph.\",\n",
+    "                },\n",
+    "                \"answer\": {\n",
+    "                    \"type\": \"string\",\n",
+    "                    \"description\": \"The answer to the question using the information in the given paragraph.\",\n",
+    "                },\n",
+    "            },\n",
+    "            \"required\": [\"question\", \"answer\"],\n",
+    "        },\n",
+    "    }\n",
     "\n",
-    "print(queries[0])"
+    "    response = openai.ChatCompletion.create(\n",
+    "        messages=messages,\n",
+    "        model=\"gpt-3.5-turbo\",\n",
+    "        functions=[submit_function],\n",
+    "        function_call=\"auto\",\n",
+    "        temperature=0.0,\n",
+    "        seed=SEED,\n",
+    "    )\n",
+    "    try:\n",
+    "        func_args = json.loads(response[\"choices\"][0][\"message\"][\"function_call\"][\"arguments\"])\n",
+    "        question = func_args[\"question\"]\n",
+    "        answer = func_args[\"answer\"]\n",
+    "        return question, answer\n",
+    "    except Exception as e:\n",
+    "        return str(e), \"N/A\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "844b40f2-533f-4ee5-9162-0091935cb8fd",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "queries = []"
```

Comment:
nit: can we put this in the cell below

Reply:
I wanted to make the cell below re-runable, without resetting the `queries=[]`.